libs/deer/tests/test_impls_core_integral.rs

```diff
@@ -1,76 +1,154 @@
+                    let deserializer = $deserializer::new(value, &context);
+
+                    let expected = $primitive::try_from(value);
+                    let received = $primitive::deserialize(deserializer);
+
+                    match expected {
+                        Ok(expected) => {
+                            // we could convert value to $primitive, therefore we can
+                            // assume that the value should also correctly deserialize
+                            let received = received.expect("should be able to deserialize");
+
+                            assert_eq!(received, expected);
+                        },
+                        Err(_) => {
+                            // the value wasn't able to fit, we should expect an error
+                            // in that case the error should be a `ValueError`
+                            // for we do not test if that is the case
+                            let _error = received.expect_err("should have not accepted the value");
+                        }
+                    }
+                }
+            }
+        }
+    };
+}
 
-    #[test]
-    fn i128_ok(value in any::<i128>()) {
-        assert_tokens(&value, &[Token::I128(value)]);
-    }
+macro_rules! proptest_integral {
+    (
+        Token::
+        $token:ident($primitive:ident);
+        $($method:ident !($($val:ident:: $visit:ident),*);)*
+    ) => {
+        paste::paste! {
+            #[cfg(not(miri))]
+            proptest! {
+                #[test]
+                fn [< $primitive _ok >](value in any::<$primitive>()) {
+                    assert_tokens(&value, &[Token::$token(value.into())])
+                }
+            }
+        }
 
-    #[test]
-    fn isize_ok(value in any::<isize>()) {
-        assert_tokens(&value, &[Token::ISize(value)]);
-    }
+        $($($method!($primitive | $val :: $visit);)*)*
+    };
 }
 
+proptest_integral!(
+    Token::Number(u8);
+    proptest_try_fit!(i8 :: I8Deserializer, i16 :: I16Deserializer, i32 :: I32Deserializer, i64 :: I64Deserializer, i128 :: I128Deserializer, u16 :: U16Deserializer, u32 :: U32Deserializer, u64 :: U64Deserializer, u128 :: U128Deserializer);
+);
+
+proptest_integral!(
+    Token::Number(u16);
+    proptest_fit!(u8 :: U8Deserializer);
+    proptest_try_fit!(i8 :: I8Deserializer, i16 :: I16Deserializer, i32 :: I32Deserializer, i64 :: I64Deserializer, i128 :: I128Deserializer, u32 :: U32Deserializer, u64 :: U64Deserializer, u128 :: U128Deserializer);
+);
+
+proptest_integral!(
+    Token::Number(u32);
+    proptest_fit!(u8 :: U8Deserializer, u16 :: U16Deserializer);
+    proptest_try_fit!(i8 :: I8Deserializer, i16 :: I16Deserializer, i32 :: I32Deserializer, i64 :: I64Deserializer, i128 :: I128Deserializer, u64 :: U64Deserializer, u128 :: U128Deserializer);
+);
+
+proptest_integral!(
+    Token::Number(u64);
+    proptest_fit!(u8 :: U8Deserializer, u16 :: U16Deserializer, u32 :: U32Deserializer);
```

Comment:
Hmm, i see we are proptesting u8 here even through we didn't specify them in `num_from` for u64. Is the u8 impl defaulted somehow?

Reply:
huh, I mean it shouldn't so that is very interesting

Reply:
Oooo, this works because `visit_u8` default implementation defers to `visit_number`, which is implemented, but I wouldn't say that this was my intention

Reply:
fixed in [fc2d6bb](https://github.com/hashintel/hash/pull/2423/commits/fc2d6bb6adfc3c69e982b6723b3f3927c3cb0bff)